packages/e2e-tests/specs/experiments/template-part-insertion.test.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	createNewPost,
+	insertBlock,
+	disablePrePublishChecks,
+} from '@wordpress/e2e-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import {
+	enableExperimentalFeatures,
+	disableExperimentalFeatures,
+} from '../../experimental-features';
+import { trashExistingPosts } from '../../config/setup-test-framework';
+
+describe( 'Template part insertion by placeholder block', () => {
+	// Test constants for template part.
+	const testSlug = 'test-template-part';
+	const testTheme = 'test-theme';
+	const testContent = 'some words...';
+
+	// Selectors
+	const chooseButtonSelector =
+		'//div[contains(@class,"is-selected")]//button[text()="Choose"]';
+	const entitiesSaveSelector = '.editor-entities-saved-states__save-button';
+	const savePostSelector = '.editor-post-publish-button__button';
+	const templatePartSelector = '*[data-type="core/template-part"]';
+	const activatedTemplatePartSelector = `${ templatePartSelector } .block-editor-inner-blocks`;
+	const templatePartButtonSelector = `${ templatePartSelector } button`;
+	const testContentSelector = `//p[contains(., "${ testContent }")]`;
+
+	// Setup & Teardown.
+	const requiredExperiments = [ '#gutenberg-full-site-editing' ];
+	beforeAll( async () => {
+		await enableExperimentalFeatures( requiredExperiments );
+		await trashExistingPosts( 'wp_template' );
+		await trashExistingPosts( 'wp_template_part' );
+	} );
+	afterAll( async () => {
+		await disableExperimentalFeatures( requiredExperiments );
```

Comment:
Should we trash the test template and template part at teardown?

```suggestion
		await disableExperimentalFeatures( requiredExperiments );
		await trashExistingPosts( 'wp_template' );
		await trashExistingPosts( 'wp_template_part' );
```

Reply:
I wondered the same thing.  But it seems the standard in these e2e tests in general is trashing in `beforeAll` and not after.  The `beforeAll` [defined to run for every test suite](https://github.com/WordPress/gutenberg/blob/2f832648001db19cae157ef4f07f39896fa6ee2a/packages/e2e-tests/config/setup-test-framework.js#L286) uses the standard `trashExistingPosts`, and that function is neglected from the afterAll defined beneath it.   I guess it would be more complete to do it after as well, but probably not necessary as we are mostly concerned with clearing the slate for the tests to run on?

Reply:
Sounds convincing, especially if we're following an established pattern here :+1: